edx-solutions/xblock-drag-and-drop-v2
# Add a checkbox to the zone tab to control visibility of zone borders.…

# Add a checkbox to the zone tab to control visibility of zone borders.  Render the borders accordingly.

## Issues addressed
This change addresses:
* https://openedx.atlassian.net/browse/SOL-1405
* https://tasks.opencraft.com/browse/OC-1161.

## User-visible changes

* This screenshot shows the newly-added "Zone borders" checkbox control in the Drag and Drop editor:
![oc-1161-0](https://cloud.githubusercontent.com/assets/6389399/12178808/22c2e53a-b57c-11e5-8a9f-32147a11f7d3.png)

* This screenshot shows the newly-added borders of the drop-zones displayed to a student (when the checkbox is ticked):
![oc-1161-1](https://cloud.githubusercontent.com/assets/6389399/12178812/28b7d0c2-b57c-11e5-8890-1627765ca8a7.png)

## Test prodedure
1. Sign in to Studio as as the "staff@example.com" user.
2. Click on "edX Demonstration Course". This opens the course outline page.
3. Scroll to "Example Week 1: Getting Started".  Click and expand it if necessary.
4. Select a subsection, e.g. " Homework - Question Styles".
5. Click "New Unit".  Give the new unit a name: "Test drag and drop borders"
6. Click "Advanced" → "Drag and drop".  The drag and drop XBlock loads.
7. Click "Continue".  The zones pane of the wizard loads.
8. Scroll down.  **Confirm that the "Zone borders" checkbox is displayed**.  Tick it.
9. Click "Continue" → "Save".  The XBlock is displayed again.
10. **Confirm that the borders of the zones appear, as in the second screenshot above.**
11. Click "Edit" and repeat steps 1-10, this time with the box unticked.
12. **Confirm that the borders of the zones no longer appear.**
13. Click "Preview".  **Confirm that the XBlock displays correctly in LMS.**

## Legal
This PR is covered by OpenCraft's contributor agreement
